docs/adrs/0010-jetstream-shared-schema.md

```diff
@@ -0,0 +1,155 @@
+# Jetstream/Nimbus Shared Schema Architecture
+
+* Status: proposed
+* Deciders: Mike Williams
+* Date: 2023-04-21
+
+Technical Story: https://mozilla-hub.atlassian.net/browse/EXP-2265
+
+## Context and Problem Statement
+
+Experiment analysis for the Nimbus experimentation platform occurs in a separate project called Jetstream. Jetstream delivers results to Experimenter, which must parse these results based on the expected format of Jetstream’s output. However, Experimenter is forced to make assumptions about what it expects in the results, increasing the potential for buggy behavior during both ingestion and on the Results UI for Experimenter. Currently, various schemas are defined in a few different places for different purposes, but one schema definition shared between Experimenter and Jetstream would allow for validation of outputs from Jetstream while removing the need for Experimenter to make assumptions about the data it is receiving from Jetstream. This document describes the decisions around how and where that schema will be defined.
+
+
+## Decision Drivers
+
+* Single schema definition shared by Experimenter and Jetstream
```

Comment:
Will we only have the jetstream result schema in there?

Jetstream also consumes data from the Experimenter API and would also benefit from having a schema that represents the API

Reply:
Yea I think that is a natural follow-on to this. It should be fairly simple to use the same decisions made here and the resulting setup for related schemas, and that one should be even easier because it is the same components consuming it.

Reply:
Oh, I see where I put it -- it's mentioned in the Pros section for Option B towards the bottom. Let me know if you think it's worth clarifying up front though and I can add a note.

Reply:
@scholtzan that schema already exists, it's published in the [nimbus-shared](https://github.com/mozilla/nimbus-shared) repo and consumable via [pypi](https://pypi.org/project/mozilla-nimbus-shared/).  The Experimenter V6 API that jetstream consumes validates against this schema, [example here](https://github.com/mozilla/experimenter/blob/main/experimenter/experimenter/experiments/tests/api/v6/test_serializers.py#L112)

I hope that after @mikewilli gets through this body of work we can use this as a foundation to move and consolidate all of the schemas for Nimbus into the Experimenter repo and publish them together into a Nimbus schemas package with bindings for Python, TypeScript, and Rust that can be consumed by all the parts of the Nimbus system.